### DISCUSSION TITLE
Setting custom fields inside auth token

### DISCUSSION THREAD

**Discussion by shanuflash** [pocketbase/pocketbase#6179] [Q&A]
Hello, am trying to see if its possible to modify the token I get while using `authWithPassword()` 
```
{
  "collectionId": "_pb_users_auth_",
  "exp": 1735735062,
  "id": "c9p62rilsesp432",
  "refreshable": true,
  "type": "auth"
}
 ```
 
 Is it possible to add say `role` from my user table?

**Comment by ganigeorgiev** [comment]
The JWT payload fields has no relation to the auth collection fields and they cannot be changed (_at least not out of the box_). You can create a custom token in a similar way to [`record.NewAuthToken()` helper](https://github.com/pocketbase/pocketbase/blob/v0.23.12/core/record_tokens.go#L51-L73) and return it as part of the [`OnRecordAuthRequest` hook](https://pocketbase.io/docs/go-event-hooks/#onrecordauthrequest) but without details about your use case and how/for what do you plan to use that field, I'd advice against it.

Note that even if you add a custom field in the token you can't blindly trust it and will still have to send refresh or `getOne` request to retrieve the user data (_the value of the field could also change after issuing the token_) so for me it doesn't really make much sense.
